src/test/java/com/commercetools/sync/sdk2/categories/CategorySyncOptionsBuilderTest.java

```diff
@@ -0,0 +1,282 @@
+package com.commercetools.sync.sdk2.categories;
+
+import static com.commercetools.api.models.common.LocalizedString.ofEnglish;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+import com.commercetools.api.client.ProjectApiRoot;
+import com.commercetools.api.models.category.*;
+import com.commercetools.api.models.common.LocalizedString;
+import com.commercetools.sync.sdk2.commons.exceptions.SyncException;
+import com.commercetools.sync.sdk2.commons.utils.QuadConsumer;
+import com.commercetools.sync.sdk2.commons.utils.TriConsumer;
+import com.commercetools.sync.sdk2.commons.utils.TriFunction;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.function.Function;
+import org.junit.jupiter.api.Test;
+
+class CategorySyncOptionsBuilderTest {
+  private static final ProjectApiRoot CTP_CLIENT = mock(ProjectApiRoot.class);
+  private CategorySyncOptionsBuilder categorySyncOptionsBuilder =
+      CategorySyncOptionsBuilder.of(CTP_CLIENT);
+
+  @Test
+  void of_WithClient_ShouldCreateCategorySyncOptionsBuilder() {
+    final CategorySyncOptionsBuilder builder = CategorySyncOptionsBuilder.of(CTP_CLIENT);
+    assertThat(builder).isNotNull();
+  }
+
+  @Test
+  void build_WithClient_ShouldBuildCategorySyncOptions() {
+    final CategorySyncOptions categorySyncOptions = categorySyncOptionsBuilder.build();
```

Comment:
It would be better if you create the builder here so that we know it's with the client only.

```suggestion
  final CategorySyncOptionsBuilder categorySyncOptionsBuilder =
      CategorySyncOptionsBuilder.of(CTP_CLIENT);
    final CategorySyncOptions categorySyncOptions = categorySyncOptionsBuilder.build();
```